### PR TITLE
Upgrade library

### PR DESCRIPTION
**Pull request by Frikki**
This PR adds a `PseudorandomNumberGenerator`  interface to enable users to hide library dependency and avoid leaking implementation details. The PR also rewrites the API to reflect the true nature of the library, namely that it is _pseudorandom_.

In addition, it removes `yarn` dependency, upgrades to use prettier with ESLint, adds pre-commit and pre-push hooks, upgrades Travis configuration to use the latest Node.js LTS.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{
+    "bracketSpacing": true,
+    "jsxBracketSameLine": false,
+    "parser": "typescript",
+    "printWidth": 80,
+    "semi": false,
+    "singleQuote": true,
+    "tabWidth": 4,
```
Comment (TylorS) [inline .prettierrc:8]: Can you humour me and make this `2`? 🙃

File: package.json [hunk 2]
```diff
@@ -1,49 +1,60 @@
...
   "name": "pure-number-generator",
-  "description": "Purely functional random number generation",
-  "repository": "https://github.com/TylorS/pure-number-generator",
   "version": "1.1.0",
+  "description": "Purely functional pseudorandom number generation",
+  "repository": "https://github.com/TylorS/pure-number-generator",
+  "license": "MIT",
+  "author": "Tylor Steinberger <tlsteinberger167@gmail.com>",
+  "contributors": [
+    "Frederik Krautwald <fkrautwald@gmail.com>"
```
Comment (TylorS) [inline package.json:9]: 🎉

File: package.json [hunk 3]
```diff
@@ -1,49 +1,60 @@
...
-    "test:unit": "./node_modules/.bin/test -r ts-node/register 'src/**/*.test.ts'"
+    "fix-ts": "eslint --fix './src/*.ts'",
+    "test": "npm run test:lint && npm run test:unit",
+    "test:lint": "eslint './src/*.ts'",
+    "test:unit": "typed-test 'src/**/*.__test__.ts'",
+    "preversion": "npm run test && npm run build",
+    "postversion": "npm run changelog && git push origin master --tags && npm publish"
+  },
+  "dependencies": {
+    "167": "^0.43.0"
```
Comment (TylorS) [inline package.json:35]: Oh wow, this is old 🙃

File: src/PseudorandomNumberGenerator.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+export interface PseudorandomNumberGenerator {
```
Comment (TylorS) [inline src/PseudorandomNumberGenerator.ts:1]: 👍 For extracting into an interface

File: src/PseudorandomNumberGenerator.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    readonly seeds: () => [number, number, number, number]
+}
+
+export class PseudorandomNumberGeneratorImpl
```
Comment (TylorS) [inline src/PseudorandomNumberGenerator.ts:5]: ❓ What do you think about ditching the class all together and just having a `createPsuedorandomNumberGenerator(0, 1, 2, 3)`?

File: src/next.ts [hunk 6]
```diff
@@ -1,22 +1,29 @@
-import { RandomNumberGenerator } from './RandomNumberGenerator'
+import {
+    PseudorandomNumberGenerator,
+    PseudorandomNumberGeneratorImpl,
+} from './PseudorandomNumberGenerator'
 
 const REAL_DIVISOR = 4294967296.0
+const FIRST_AVALANCHE_ROTATION = 27
```
Comment (TylorS) [inline src/next.ts:7]: 🎉 For these variables. I'd never know what to call this

**Review by TylorS — APPROVED** [review]

File: src/PseudorandomNumberGenerator.ts (hunk 5 above)
Comment (Frikki) [reply src/PseudorandomNumberGenerator.ts:5]: Let me think about that. In any case, it is for another PR.

File: src/next.ts (hunk 6 above)
Comment (Frikki) [reply src/next.ts:7]: By reading the BurtlePRNG and the thorough description of the algorithm, I figured out why these numbers, what they do, and what they are.

**Pull request merged**